StoreTestWPF.ViewModel/ViewModels/StoreViewModel.cs

```diff
@@ -0,0 +1,206 @@
+﻿using StoreTestWPF.DAL;
+using StoreTestWPF.DAL.Models;
+using StoreTestWPF.ViewModel.Interfaces;
+using StoreTestWPF.ViewModel.Utils;
+using System.Collections.ObjectModel;
+using System.Data.Entity;
+using System.Linq;
+using System.Windows.Input;
+
+namespace StoreTestWPF.ViewModel.ViewModels
+{
+    public sealed class StoreViewModel : ViewModelBase
+    {
+        private CakeViewModel selectedCake;
+        private bool isCakeSelected;
+        private ICommand loadCommand;
+        private ICommand addCommand;
+        private ICommand editCommand;
+        private ICommand deleteCommand;
+        private IViewService viewService;
+        private CakeStoreDbContext dbContext;
+
+        public ObservableCollection<CakeViewModel> Cakes { get; private set; }
+
+        public CakeViewModel SelectedCake
+        {
+            get { return this.selectedCake; }
+            set
+            {
+                if (this.selectedCake != value)
+                {
+                    this.selectedCake = value;
+                    OnPropertyChanged(nameof(this.SelectedCake));
+                    OnPropertyChanged(nameof(this.IsCakeSelected));
+                }
+            }
+        }
+
+        public bool IsCakeSelected
+        {
+            get
+            {
+                return !(this.selectedCake == null);
+            }
+            set
+            {
+                if (this.isCakeSelected != value)
+                {
+                    this.isCakeSelected = value;
+                    OnPropertyChanged(nameof(this.IsCakeSelected));
+                }
+            }
+        }
+
+        public ICommand LoadCommand
+        {
+            get
+            {
+                return this.loadCommand ?? new RelayCommand(LoadExecuted, LoadCanExecute);
+            }
+        }
+
+        public ICommand AddCommand
+        {
+            get
+            {
+                return this.addCommand ?? new RelayCommand(AddExecuted);
+            }
+        }
+
+        public ICommand EditCommand
+        {
+            get
+            {
+                return this.editCommand ?? new RelayCommandWithSender<CakeViewModel>(EditExecuted, EditCanExecute);
+            }
+        }
+
+        public ICommand DeleteCommand
+        {
+            get
+            {
+                return this.deleteCommand ?? new RelayCommandWithSender<CakeViewModel>(DeleteExecuted, DeleteCanExecute);
+            }
+        }
+
+        public StoreViewModel(IViewService viewService)
+        {
+            this.Cakes = new ObservableCollection<CakeViewModel>();
+            this.dbContext = new CakeStoreDbContext();
+            this.viewService = viewService;
+        }
+
+        private void AddExecuted()
+        {
+            ModifyCakeViewModel modifyViewModel = new ModifyCakeViewModel(this.viewService);
```

Comment:
Lets try using 'var' in cases when we create instance using constructor.

Reply:
Fixed